convert2rhel/unit_tests/cert_test.py

```diff
@@ -70,20 +77,32 @@ def test_get_cert_path_missing_cert(self):
     @unit_tests.mock(utils, "DATA_DIR", unit_tests.NONEXISTING_DIR)
     @unit_tests.mock(system_info, "arch", "nonexisting_arch")
     def test_get_cert_path_nonexisting_dir(self):
-        self.assertRaises(SystemExit, cert.SystemCert._get_cert_path)
+        self.assertRaises(SystemExit, cert.SystemCert._get_cert)
         self.assertEqual(len(cert.loggerinst.critical_msgs), 1)
 
-    @unit_tests.mock(cert.SystemCert, "_system_cert_dir", unit_tests.TMP_DIR)
     @unit_tests.mock(tool_opts, "arch", "x86_64")
     @unit_tests.mock(utils, "DATA_DIR", os.path.join(base_data_dir, "6", "x86_64"))
     def test_install_cert(self):
         # By initializing the cert object we get a path to an existing
         # certificate based on the mocked parameters above
         system_cert = cert.SystemCert()
+        system_cert._target_cert_dir = unit_tests.TMP_DIR
 
         system_cert.install()
 
-        cert_filename = os.path.basename(system_cert._cert_path)
-        installed_cert_path = os.path.join(system_cert._system_cert_dir, cert_filename)
+        installed_cert_path = os.path.join(system_cert._target_cert_dir, system_cert._cert_filename)
         self.assertTrue(os.path.exists(installed_cert_path))
         shutil.rmtree(unit_tests.TMP_DIR)
+
+
+def test_remove(monkeypatch):
+    os_remove = mock.Mock()
+    monkeypatch.setattr(os, "remove", value=os_remove)
+
+    _get_cert = mock.Mock(return_value=("anything", "anything"))
+    monkeypatch.setattr(cert.SystemCert, "_get_cert", value=_get_cert)
+
+    sys_cert = cert.SystemCert()
+    sys_cert.remove()
+
+    os_remove.assert_called_once()
```

Comment:
This test is not that useful. You mocking the main piece of the functionality - removing the file. That would be fine if you would like to test if logger messages were emitted. Otherwise, it is better to mock as minimum things as possible.
The goal of your function to remove the target cert path. So instead of mocking the remove function, try to play with the value of the cert path. Check if it is None, not existing file, and existing file

Reply:
@ZhukovGreen Fixed, now it tests real remove of file if it exists, otherwise it raise exception (and check, if logger.error was called). I think, that variant with None would crash earlier, than it would come to remove method. But I put it there :)

Reply:
Lately we've been using caplog, it's easier. E.g. https://github.com/oamg/convert2rhel/blob/main/convert2rhel/unit_tests/subscription_test.py#L432.